msteams-platform/messaging-extensions/high-quality-message-extension.md

```diff
@@ -708,25 +867,97 @@ Message extensions respond to a user input with an Adaptive Card. An Adaptive Ca
+  :::image type="content" source="../assets/images/Copilot/ailib-copilot-action-buttons.png" alt-text="Graphic shows an example of the Update Stock, restock, and Cancel restock action buttons in an Adaptive Card response in Copilot for Microsoft 365 in Teams.":::
+
+* Adaptive Cards must not display a horizontal scroll. To avoid horizontal scrolls, don’t specify a fixed width. *[Mandatory fix]*
+
+  * **ColumnSets**
+
+    * Don't define `ColumnSets` with more than three columns.
+    * Don’t use explicit pixel width on more than one column in the set.
+    * Ensure the column doesn't exceed one-quarter of the narrowest card width, such as in a meeting chat or Copilot.
+    * Generally, an explicit width must not exceed 48 pixels, though some scenarios might allow for exceptions.
+
+  * **Sizing images**
+  
+    * When using an image inside a `ColumnSet` with more than one Column, specify the size of the column containing an image rather than the image itself.
+    * If the image isn’t in a `ColumnSet`, we recommend you to set its size to `auto` or `stretch`.
+    * If you want to define explicit width in pixels, ensure that they don't exceed 3/4 of the narrowest card width.
+    * If you want to define explicit size in pixels, define it for the width or height. Setting explicit size for any one parameter preserves the image's aspect ratio.
+    * We recommend you to set the width of the image, though some scenarios might allow for exceptions.
 
 * If a user can change any information on the card through dialog, Stageview, or directly from the card, we recommend the Adaptive Card to support universal actions and automatic refresh. [*Recommended*]
 * Adaptive Cards must include a URL as part of the [metadata](https://adaptivecards.io/explorer/Metadata.html), which allows cards to be easily copied from one hub to another. [*Recommended*]
 * Apart from thumbnails, any image in an Adaptive Card must have an alt-text. [*Recommended*]
 
+## Extend your plugin to Copilot
+
+Copilot for Microsoft 365 is available in Teams, Outlook, and Bing. To enable your message extensions as a plugin in Copilot for Microsoft 365, we recommend you to implement the following best practices:
+
+* **Upgrade TeamsJS Version**: Upgrade your [TeamsJS version to 2.19.0](https://www.npmjs.com/package/@microsoft/teams-js) by installing the `@microsoft/teams-js` package from npm. This ensures that your application continues to work with Bing domains. *[Mandatory fix]*
+
+* **Update Microsoft Entra ID app registration for SSO**: Microsoft Entra ID single sign-on (SSO) for message extensions works similarly for Teams, Outlook, and Bing. However, you must add client application identifiers to your bot's Microsoft Entra ID app registration in your tenant's App registrations portal. To do this, follow these steps: *[Optional]*
```

Comment:
@v-ypalikila, Why is this marked optional ? For apps with SSO, i thought it was mandatory. Please confirm this and update if needed.